app/tasks/schedule.php

```diff
@@ -0,0 +1,155 @@
+                }
+            }
+        }
+    };
+
+
+    $dbForConsole = getConsoleDB();
+    $count = 0;
+    $limit = 50;
+    $sum = $limit;
+    $functions = [];
+    $queue = [];
+
+    /**
+     * Initial run fill $functions list
+     */
+    while ($sum === $limit) {
+        $results = $dbForConsole->find('schedules', [
+            Query::equal('region', [App::getEnv('_APP_REGION')]),
+            Query::equal('type', ['function']),
+            Query::equal('active', [true]),
+            Query::limit($limit)
+        ]);
+
+        $sum = count($results);
+        foreach ($results as $document) {
+            $functions[$document['scheduleId']] = $document;
+            $count++;
+        }
+    }
+
+    $createQueue();
+    $lastUpdate =  DateTime::addSeconds(new \DateTime(), -FUNCTION_VALIDATION_TIMER);
+
+    Co\run(
+        function () use ($removeFromQueue, $createQueue, $dbForConsole, &$functions, &$queue, &$lastUpdate) {
+            Timer::tick(FUNCTION_VALIDATION_TIMER * 1000, function () use ($removeFromQueue, $createQueue, $dbForConsole, &$functions, &$queue, &$lastUpdate) {
+                $time = DateTime::now();
+                $count = 0;
+                $limit = 50;
+                $sum = $limit;
+
+                Console::info("Update proc run at: $time last update was at $lastUpdate");
+                /**
+                 * Updating functions list from DB.
+                 */
+                while ($sum === $limit) {
+                    $results = $dbForConsole->find('schedules', [
+                        Query::equal('region', [App::getEnv('_APP_REGION')]),
+                        Query::equal('type', ['function']),
+                        Query::greaterThan('scheduleUpdatedAt', $lastUpdate),
+                        Query::limit($limit)
+                    ]);
+                    $sum = count($results);
+                    foreach ($results as $document) {
+                        $org = isset($functions[$document['scheduleId']]) ? strtotime($functions[$document['scheduleId']]['scheduleUpdatedAt']) : null;
+                        $new = strtotime($document['scheduleUpdatedAt']);
+                        if ($document['active'] === false) {
+                            Console::error("Removing:  {$document['scheduleId']}");
+                            unset($functions[$document['scheduleId']]);
+                        } elseif ($new > $org) {
+                            Console::error("Updating:  {$document['scheduleId']}");
+                            $functions[$document['scheduleId']] =  $document;
+                        }
+                        $removeFromQueue($document['scheduleId']);
+                        $count++;
+                    }
+                }
+
+                $lastUpdate = DateTime::now();
+                $createQueue();
+            });
+
+            Timer::tick(FUNCTION_ENQUEUE_TIMER * 1000, function () use ($dbForConsole, &$functions, &$queue) {
```

Comment:
Not sure what is the responsibility of this loop. Why is it running every 5 minutes where the interval from crons should be at exactly 1 minute.